cmd/root.go

```diff
@@ -242,13 +251,24 @@ func (r *Root) execute() {
 		Example: rootExamples,
 	}
 	rootCmd.CompletionOptions.DisableDefaultCmd = true
+
 	rootCmd.SetHelpTemplate(rootCustomHelpTemplate)
 
 	rootCmd.PersistentFlags().BoolVar(&debugMode, "debug", false, "Run in debug mode")
 
-	// Manually parse flags to determine debug mode early
+	rootCmd.PersistentFlags().Bool("version", false, "Fetch the latest version")
```

Comment:
Why is version flag added as PersistentFlag..? PersistentFlag is a flag which can be used in other commands also I guess. But we require --version flag only for root command right?

Reply:
My approach for using a PersistentFlag here was to make it accessible not only for the root command but also for any subcommands that might need to access or check the version flag. Since you have made it clear that isn't the case , we can change the PersistantFlags() to Flags() to make it accessible only to the root command